app/analysis/index/controller.js

```diff
@@ -462,4 +460,69 @@ export default class AnalysisController extends Controller.extend(
       Object.keys(this.allQueryParams).filter((k) => k !== "ordering")
     );
   }
+
+  get allQueryParams() {
+    const queries = {};
+    for (const param of this.queryParams) {
+      // the type of param is string of object that have one key
+      if (typeof param === "string") {
+        queries[param] = this[param];
+        continue;
+      }
+      const key = Object.keys(param)[0];
+      queries[key] = this[key];
+    }
+    return queries;
+  }
+
+  get queryParamsState() {
+    const states = {};
+    for (const param of this.queryParams) {
+      // the type of param is string of object that have one key
+      const key = typeof param === "string" ? param : Object.keys(param)[0];
+      const defaultValue = this.getDefaultValue(key);
+      const currentValue = this[key];
+      states[key] = {
+        as: key,
+        defaultValue,
+        serializedValue: currentValue,
+        value: currentValue,
+        changed: currentValue !== defaultValue,
+      };
+      if (["fromDate", "toDate"].includes(key)) {
+        states[key].serialize = serializeMoment;
+        states[key].deserialize = deserializeMoment;
+      }
+      if (key === "ordering") {
+        states[key].serialize = (val) => {
+          return `${val},id`;
+        };
+        states[key].deserialize = (val) => {
+          return val.replace(",id", "");
+        };
+      }
+    }
+    return states;
+  }
+
+  getDefaultValue(paramKey) {
+    for (const param of this.queryParams) {
```

Comment:
This method seems a little useless. It's returning `undefined` and `Object.values(param)[0];` for what reason?

Reply:
May [trackedReset](https://github.com/tracked-tools/tracked-toolbox#trackedreset) could be helpful here.